hibernate-core/src/main/java/org/hibernate/boot/registry/selector/internal/DefaultDialectSelector.java

```diff
@@ -112,75 +89,85 @@ public Class<? extends Dialect> resolve(final String name) {
 			case "MariaDB":
 				return MariaDBDialect.class;
 			case "MariaDB53":
-				return MariaDB53Dialect.class;
 			case "MariaDB10":
-				return MariaDB10Dialect.class;
 			case "MariaDB102":
-				return MariaDB102Dialect.class;
+				return findCommunityDialect( name );
 			case "MariaDB103":
 				return MariaDB103Dialect.class;
 			case "MySQL":
 				return MySQLDialect.class;
 			case "MySQL5":
-				return MySQL5Dialect.class;
 			case "MySQL55":
-				return MySQL55Dialect.class;
+				return findCommunityDialect( name );
 			case "MySQL57":
 				return MySQL57Dialect.class;
 			case "MySQL8":
 				return MySQL8Dialect.class;
 			case "Oracle":
 				return OracleDialect.class;
 			case "Oracle8i":
-				return Oracle8iDialect.class;
 			case "Oracle9i":
-				return Oracle9iDialect.class;
 			case "Oracle10g":
-				return Oracle10gDialect.class;
+				return findCommunityDialect( name );
 			case "Oracle12c":
 				return Oracle12cDialect.class;
 			case "PostgresPlus":
 				return PostgresPlusDialect.class;
 			case "PostgreSQL":
 				return PostgreSQLDialect.class;
 			case "PostgreSQL81":
-				return PostgreSQL81Dialect.class;
 			case "PostgreSQL82":
-				return PostgreSQL82Dialect.class;
 			case "PostgreSQL9":
-				return PostgreSQL9Dialect.class;
 			case "PostgreSQL91":
-				return PostgreSQL91Dialect.class;
 			case "PostgreSQL92":
-				return PostgreSQL92Dialect.class;
 			case "PostgreSQL93":
-				return PostgreSQL93Dialect.class;
 			case "PostgreSQL94":
-				return PostgreSQL94Dialect.class;
 			case "PostgreSQL95":
-				return PostgreSQL95Dialect.class;
+				return findCommunityDialect( name );
+			case "PostgreSQL10":
+				return PostgreSQL10Dialect.class;
 			case "Spanner":
 				return SpannerDialect.class;
 			case "SQLServer":
 				return SQLServerDialect.class;
 			case "SQLServer2005":
-				return SQLServer2005Dialect.class;
+				return findCommunityDialect( name );
 			case "SQLServer2008":
 				return SQLServer2008Dialect.class;
 			case "SQLServer2012":
 				return SQLServer2012Dialect.class;
+			case "SQLServer2016":
+				return SQLServer2016Dialect.class;
 			case "Sybase":
 				return SybaseDialect.class;
 			case "Sybase11":
-				return Sybase11Dialect.class;
+				return findCommunityDialect( name );
 			case "SybaseASE":
 				return SybaseASEDialect.class;
 			case "SybaseASE15":
-				return SybaseASE15Dialect.class;
 			case "SybaseASE157":
-				return SybaseASE157Dialect.class;
+				return findCommunityDialect( name );
 		}
 		return null;
 	}
 
+	private static Class<? extends Dialect> findCommunityDialect(String name) {
```

Comment:
Still not entirely sure about exception v. warning here.  As I understand, the incoming `name` is a value explicitly set for which Dialect to use.

What's the ultimate outcome in this case?  We fall back to database-metadata resolution I assume?

Reply:
+1 for exception. If something gets explicitly requested and we can't find it, there's a problem that should be addressed right now.

Falling back to database-metadata resolution is unlikely to help if the user selected a community (i.e. unsupported) dialect in the first place.

Reply:
Ok, then I'll change to throwing an exception.